### PR TITLE
feature - improve code

### PR DESCRIPTION
**Pull request by at-tungnguyen**
#### What's this PR do? Please give a technical summary `(required)`
- [x] improve code

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Define/String.swift [hunk 1]
```diff
@@ -32,9 +32,24 @@ extension App {
...
+        static let FoodViewControllerTitle = "Foods"
+        static let DrinkViewControllerTitle = "Drinks"
+        static let SearchViewControllerTitle = "Search"
 
         // MARK: - Alert
         static let AddToCartAlertTitle = "Add to cart"
         static let AddToCartAlertMessage = "Your product has been added to cart."
+
+        // MARK: - Notification's Name
+        static let NotificationCart = "kNotificationManagerCart"
```
Comment (at-linhvo) [inline MyApp/Define/String.swift]: define in file Ext and use static
extension Notification.Name {
       static let kNotificationFavourite = "kNotificationFavourite"
}

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 2]
```diff
@@ -26,9 +26,17 @@ class CartViewController: ViewController {
...
-        viewModel.fetchCartItem()
-        viewModel.delegate = self
-        totalLabel.text = "\(viewModel.getTotalOrder().formattedWithSeparator()) VND"
+        setupData()
+        let name = Notification.Name(rawValue: App.String.NotificationCart)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: remove. no need function

File: MyApp/View/Controllers/Detail/DetailViewController.swift [hunk 3]
```diff
@@ -25,6 +25,16 @@ class DetailViewController: ViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
+        let name = Notification.Name(rawValue: App.String.NotificationFavorite)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Detail/DetailViewController.swift]: remove function. no need

File: MyApp/View/Controllers/Favorite/FavoriteViewController.swift [hunk 4]
```diff
@@ -14,13 +14,28 @@ class FavoriteViewController: ViewController {
...
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
         setupData()
+        let name = Notification.Name(rawValue: App.String.NotificationFavorite)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Favorite/FavoriteViewController.swift]: remove function. no need

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 5]
```diff
@@ -26,9 +26,17 @@ class CartViewController: ViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
-        viewModel.fetchCartItem()
-        viewModel.delegate = self
-        totalLabel.text = "\(viewModel.getTotalOrder().formattedWithSeparator()) VND"
+        setupData()
+        let name = Notification.Name(rawValue: Notification.Name.kNotificationCart)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: name: .kNotificationCart

File: MyApp/View/Controllers/Detail/DetailViewController.swift [hunk 6]
```diff
@@ -25,6 +25,16 @@ class DetailViewController: ViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
+        let name = Notification.Name(rawValue: Notification.Name.kNotificationFavorite)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Detail/DetailViewController.swift]: name: .kNotifi..

File: MyApp/View/Controllers/Favorite/FavoriteViewController.swift [hunk 7]
```diff
@@ -14,13 +14,28 @@ class FavoriteViewController: ViewController {
...
+    var additionalWidth: CGFloat = 75
     var viewModel = FavoriteViewModel()
 
     // MARK: - Life cycle
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
         setupData()
+        let name = Notification.Name(rawValue: Notification.Name.kNotificationFavorite)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Favorite/FavoriteViewController.swift]: name: kNotifi...

File: MyApp/View/Controllers/Home/Drink/DrinkViewController.swift [hunk 8]
```diff
@@ -16,13 +16,24 @@ class DrinkViewController: ViewController {
...
+    var additionalWidth: CGFloat = 75
     var viewModel = HomeViewModel()
 
     // MARK: - Life cycle
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
         setupData()
+        let name = Notification.Name(rawValue: Notification.Name.kNotificationFavorite)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Home/Drink/DrinkViewController.swift]: name: .kNotifi...

File: MyApp/View/Controllers/Home/Food/FoodViewController.swift [hunk 9]
```diff
@@ -16,13 +16,24 @@ class FoodViewController: ViewController {
...
+    var additionalWidth: CGFloat = 75
     var viewModel = HomeViewModel()
 
     // MARK: - Life cycle
     override func viewDidLoad() {
         super.viewDidLoad()
         setupUI()
         setupData()
+        let name = Notification.Name(rawValue: Notification.Name.kNotificationFavorite)
+        NotificationCenter.default.addObserver(self, selector: #selector(receiveNotification), name: name, object: nil)
```
Comment (at-linhvo) [inline MyApp/View/Controllers/Home/Food/FoodViewController.swift]: name: .kNotifi...

**Pull request merged**